src/ui/Omnibox/Omnibox.ts

```diff
@@ -747,7 +749,7 @@ export class Omnibox extends Component {
       return wordCompletion;
     }
 
-    return this.magicBox.getWordCompletion() || this.getFirstSuggestion() || this.magicBox.getText();
+    return this.magicBox.getWordCompletion() || this.getFirstSuggestion() || this.lastQuery || this.magicBox.getText();
```

Comment:
I am curious to know why you added `this.lastQuery` here? Is it being captured by a unit test?

Reply:
It's a very special corner case, but grosso modo, when you configure ML query suggestions + search as you type + ML confidence that the suggestion is "good" is high enough, we automatically do the query for you by using the suggestion (kinda like and instant search).

Now, when you tab out, but the query that was executed for you does not match you currently have typed in the search box, we want to "switch" the content of the input with the actual query that was performed.

The first two checks does that (return the first suggestion if its a good one), but with that change to tabpress, the suggestions are gone instantly when you tab out (before it would still show them, and you had to press tab twice).

As such I'm adding this here to recover the last query performed so we get the correct content in the input.

As for a UT, that's a good point. Although as you can see testing for the particular case im describing is a very special "corner case", and would probably be better tested in a functional test, since it involves quite a bit of orchestration of different keyboard events. Writing a functional test for this would be much better, I'll talk with Lucille.